### PR TITLE
sqlcipher: enable JSON1 extension

### PR DESCRIPTION
**Pull request by wmertens**
our sqlite has JSON1 enabled but sqlcipher doesn't. It is used in sqlitebrowser now so it should maintain parity.

### CODE REVIEW

File: pkgs/development/libraries/sqlcipher/default.nix [hunk 1]
```diff
@@ -21,9 +21,24 @@ stdenv.mkDerivation rec {
...
+    "-DSQLITE_ENABLE_FTS5"
+    "-DSQLITE_ENABLE_RTREE"
+    "-DSQLITE_ENABLE_STMT_SCANSTATUS"
+    "-DSQLITE_ENABLE_UNLOCK_NOTIFY"
+    "-DSQLITE_SOUNDEX"
+    "-DSQLITE_SECURE_DELETE"
+    "-DSQLITE_MAX_VARIABLE_NUMBER=250000"
+    "-DSQLITE_MAX_EXPR_DEPTH=10000"
+
+    # Additional flags for sqlcipher
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/sqlcipher/default.nix:41]:
```suggestion
  CFLAGS = sqlite.CFLAGS ++ [
```
Can we do this?

File: pkgs/development/libraries/sqlcipher/default.nix [hunk 2]
```diff
@@ -21,9 +21,24 @@ stdenv.mkDerivation rec {
   ];
 
   CFLAGS = [
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/sqlcipher/default.nix:23]:
```suggestion
```

**Comment by wmertens** [comment]
@SuperSandro2000 unfortunately the CFLAGS array isn't available but I saw that NIX_CFLAGS_COMPILE string is there. This may not be a great idea but it works :)

**Review by SuperSandro2000 — APPROVED** [review]
> @SuperSandro2000 unfortunately the CFLAGS array isn't available but I saw that NIX_CFLAGS_COMPILE string is there. This may not be a great idea but it works :)

and it is way better than keeping a list of ~15 strings in sync.

**Pull request merged**